src/creature.cpp

```diff
@@ -814,6 +814,11 @@ void Creature::add_effect( efftype_id eff_id, int dur, body_part bp,
                 e.mod_intensity(e.get_int_add_val());
             }
 
+            // Force intensity if it is duration based
+            if( e.get_int_dur_factor() != 0 ) {
+                // + 1 here so that the lowest is intensity 1, not 0
+                e.set_intensity( ( e.get_duration() / e.get_int_dur_factor() ) + 1 );
+            }
```

Comment:
I assumed no, because of the existing check for bad intensity. I'm compiling without this check to entry to see what happens.